src/Translatable/Translatable.php

```diff
@@ -458,4 +459,35 @@ public function __isset($key)
     {
         return $this->isTranslationAttribute($key) || parent::__isset($key);
     }
+
+    public function getTranslationChanges(): array
+    {
+        switch (config('translatable.rule_factory.format')) {
+            case RuleFactory::FORMAT_KEY:
+                $replacementFunc = function (string $key, string $locale): string {
+                    return $key . ':' . $locale;
+                };
+                break;
+            default:
+                $replacementFunc = function (string $key, string $locale): string {
+                    return $locale . '.' . $key;
+                };
+        }
+        $translationChanges = array();
+        foreach ($this->translations as $translation) {
+            $locale = $translation->locale;
+            foreach ($translation->getChanges() as $attribute => $value) {
+                $translationChanges[$replacementFunc($attribute, $locale)] = $value;
+            }
+        }
+        return $translationChanges;
+    }
+
+    public function wasTranslationChanged($attributes = null): bool
+    {
+        return $this->hasChanges(
+            $this->getTranslationChanges(),
+            is_array($attributes) ? $attributes : func_get_args()
+        );
```

Comment:
I'm not sure if the `hasChanges()` method understands dot.notation? 🤔
The return of `getTranslationChanges()` will be a multi-dimensional array.
```php
[
  'de' => [
    'foo' => 'bar',
  ]
]
```

But I would expect to be able to call `wasTranslationChanged()` like `wasTranslationChanged('foo')` (to check if 'foo'. changed in any translation) or `wasTranslationChanged('foo', 'de')` to check if 'foo' changed in german translation.

So I think that we would need custom logic here.

Reply:
As far as I understood the Laravel model's method signature, you can pass more than one argument to find out if either of the attributes has changed. So if we stick with Laravel's convention in `wasChanged()` the distinction for language as a second argument won't work.

Reply:
As it's a totally decoupled method I don't mind to be different.
But good point the signature should be `wasTranslationChanged(string|array $attribute, ?string $locale = null)`.

`Arr::dot()` could help with multidimensional array and dot.notation

Reply:
So finally it became a lot of code to keep things performant and all input options available...🤷🏻‍♂️